server/game/cards/events/05/daringrescue.js

```diff
@@ -0,0 +1,44 @@
+const DrawCard = require('../../../drawcard.js');
+
+class DaringRescue extends DrawCard {
+    setupCardAbilities() {
+        this.action({
+            title: 'Return character then have knight gain power',
+            condition: () => this.hasKnightCharacter(),
+            target: {
+                activePromptTitle: 'Select a character to return',
+                cardCondition: card => (
+                    card.location === 'play area' && 
+                    card.controller === this.controller && 
+                    card.getType() === 'character')
+            },
+            handler: context => {
+                this.controller.moveCard(context.target, 'hand');
+                this.game.promptForSelect(this.controller, {
+                    activePromptTitle: 'Select a Knight character',
+                    source: this,
+                    cardCondition: card => (
+                        card.location === 'play area' && 
+                        card.hasTrait('Knight') &&
+                        card.getType() === 'character' && 
+                        card.controller === this.controller),
+                    onSelect: (p, card) => {
+                        card.modifyPower(1);
+                        this.game.addMessage('{0} plays {1} to return {2} to its owner\'s hand then have {3} gain 1 power', 
+                                              this.controller, this, context.target, card);
```

Comment:
Well, the promptForSelect method always has a 'Done' option in case there's no legal target, although that does mean no message at all will be displayed. So you're right the message needs to be built up in multiple steps I think...